src/mlpack/tests/main_tests/local_coordinate_coding_test.cpp

```diff
@@ -0,0 +1,417 @@
+}
+
+/**
+ * Ensure that dimensionality of the trained model matches
+ * the dimensionality of the test points.
+ */
+BOOST_AUTO_TEST_CASE(LCCTrainedModelDimTest)
+{
+  arma::mat x;
+  x.load("mnist_first250_training_4s_and_9s.arm");
+  arma:: mat t = x;
+  t.shed_rows(1, 2);
+
+  SetInputParam("training", x);
+  SetInputParam("atoms", (int) 10);
+  SetInputParam("max_iterations", (int) 2);
+
+  mlpackMain();
+
+  LocalCoordinateCoding* outputModel =
+      std::move(CLI::GetParam<LocalCoordinateCoding*>("output_model"));
+
+  SetInputParam("input_model", std::move(outputModel));
+  SetInputParam("test", std::move(t));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+* Ensure that the program throws error for negative number of atoms.
+*/
+BOOST_AUTO_TEST_CASE(LCCAtomsBoundTest)
+{
+  arma::mat x = {{1, 1, 1, 1}, {2, 2, 2, 2}, {3, 3, 3, 3}, {4, 4, 4, 4}};
+  SetInputParam("training", std::move(x));
+  SetInputParam("atoms", (int) -1);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+* Ensure that the program throws error for negative regularization parameter.
+*/
+BOOST_AUTO_TEST_CASE(LCCNegativeLambdaTest)
+{
+  arma::mat x = {{1, 1, 1, 1}, {2, 2, 2, 2}, {3, 3, 3, 3}, {4, 4, 4, 4}};
+  SetInputParam("training", std::move(x));
+  SetInputParam("atoms", (int) 2);
+  SetInputParam("lambda", -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+* Ensure that the program throws error for negative tolerance.
+*/
+BOOST_AUTO_TEST_CASE(LCCNegativeToleranceTest)
+{
+  arma::mat x = {{1, 1, 1, 1}, {2, 2, 2, 2}, {3, 3, 3, 3}, {4, 4, 4, 4}};
+  SetInputParam("training", std::move(x));
+  SetInputParam("atoms", (int) 2);
+  SetInputParam("tolerance", -1.0);
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/*
+* Ensure that the normalize parameter works.
+*/
+BOOST_AUTO_TEST_CASE(LCCNormalizationTest)
+{
+  arma::mat x = {{1, 2, 3, 4}, {2, 2, 3, 1}, {3, 2, 3, 0}, {1, 1, 4, 4}};
+  arma::mat t = x;
+  arma::mat initDict = {{1, 2}, {2, 3}, {3, 4}, {4, 5}};
+
+  SetInputParam("training", x);
+  SetInputParam("atoms", (int) 2);
+  SetInputParam("initial_dictionary", initDict);
+  SetInputParam("max_iterations", 2);
+  SetInputParam("test", t);
+  SetInputParam("normalize", (bool) true);
+
+  mlpackMain();
+
+  arma::mat codes = CLI::GetParam<arma::mat>("codes");
+  arma::mat dictionary = CLI::GetParam<arma::mat>("dictionary");
+
+  bindings::tests::CleanMemory();
+  CLI::Parameters()["normalize"].wasPassed = false;
+
+  // Normalize the datasets.
+  for (size_t i = 0; i < x.n_cols; i++)
+    x.col(i) /= norm(x.col(i), 2);
+
+  for (size_t i = 0; i < t.n_cols; i++)
+    t.col(i) /= norm(t.col(i), 2);
```

Comment:
Actually for this one, we should avoid normalizing here and then test that the dictionary and codes from the two different runs are different.  The reason is that although you've specified `normalize` as false for the second run and manually normalized `x` and `t`, it's still possible that the binding might be normalizing anyway---which wouldn't change the data at all (and thus give the same results).